robot/basestation/static/js/roslibjs/ros-helpers.js

```diff
@@ -188,6 +204,76 @@ function initRosWeb () {
 }
 
 /* functions used in main code */
+
+function rosTimestamp () {
+  // next few lines taken and adjusted from roslibjs action server example
+  let currentTime = new Date()
+  let secs = currentTime.getTime()/1000 // seconds before truncating the decimal
+  let secsFloored = Math.floor(secs) // seconds after truncating
+  let nsecs = Math.round(1000000000*(secs-secsFloored)) // nanoseconds since the previous second
+
+  // return a dictionary for the ROS log and a string for the gui console
+  let stampTime = currentTime.toString().split(' ')[4] // hh:mm:ss from date object
+  return [{secs : secsFloored, nsecs : nsecs}, stampTime]
+}
+
+function publishRosLog (logLevel, timestamp, message) {
+  ros_logger.publish(
+    new ROSLIB.Message({
+      // I'm only publishing the essentials. We could include more info if so desired
+      header : {
+        // seq // uint32: sequence ID, seems to increment automatically
+        stamp : timestamp // dictionary: contains truncated seconds and nanoseconds
+        // frame_id // string: probably only useful for tf
+      },
+      level : logLevel, // int: see log level constants above
+      // name : '/web_gui', // name of the node (proposed)
+      msg : message, // string: this is the log message
+      // file // string: we could specify the js file that generated the log
+      // function // string: we could specify the parent function that generated the log
+      // line // uint32: we could specify the specific line of code that generated the log
+      // topics // string[]: topic names that the node publishes
+    })
+  )
+}
+
+function rosLog (logLevel, message) {
+  logData = {}
+  logData[ROSDEBUG] = {prefix : '[DEBUG]', isError : false}
+  logData[ROSINFO] = {prefix : '[INFO]', isError : false}
+  logData[ROSWARN] = {prefix : '[WARN]', isError : true}
```

Comment:
lets do two things:
1) since semantically (and like in C++) warnings != erros this `isError` should be `false`
2) we should use `console.warn('something')`

Reply:
dealt with